### PR TITLE
Fix polledExpect

### PR DESCRIPTION
**Pull request by sharp-bits**
The wait loop wasn't synchronized. In just the unit test, this resulted in around 2k asyncblock tasks added to the flow. In production, it was over 100k. This was enough to run the node process out of memory (sometimes).

### CODE REVIEW

**Comment by scriby** [comment]
Cool! Glad you found the source.

Kind of an embarrassing mistake coming from the author of asyncblock, but we all make mistakes, right :)?

**Comment by sharp-bits** [comment]
At first, it looked like there was a nasty leak in asyncblock because it
tracks finished tasks (intentional memory leak?). Pretty obvious that
creating 120k tasks in a few seconds wasn't a planned use case.

**Comment by shermaneric** [comment]
Thanks all for the investigation and help!
Just for my own learning, taking this fork - verified removing the flow.wait() makes this test fail.
I also locally added in Mindy's investigatory tests [here](https://github.com/blackboard/protractor-sync/pull/35) and looped through her heap test 10 times.

Looks good!

With the flow.wait:
```
-rw-r--r--    1 esherman  staff  15795799 Jul 25 15:09 a_beforeFailure.heapsnapshot
-rw-r--r--    1 esherman  staff  15684789 Jul 25 15:08 afterFailure0.heapsnapshot
-rw-r--r--    1 esherman  staff  15631757 Jul 25 15:09 afterFailure1.heapsnapshot
-rw-r--r--    1 esherman  staff  15652179 Jul 25 15:09 afterFailure2.heapsnapshot
-rw-r--r--    1 esherman  staff  15660711 Jul 25 15:09 afterFailure3.heapsnapshot
-rw-r--r--    1 esherman  staff  15666734 Jul 25 15:09 afterFailure4.heapsnapshot
-rw-r--r--    1 esherman  staff  15672878 Jul 25 15:09 afterFailure5.heapsnapshot
-rw-r--r--    1 esherman  staff  15687200 Jul 25 15:09 afterFailure6.heapsnapshot
-rw-r--r--    1 esherman  staff  15694909 Jul 25 15:09 afterFailure7.heapsnapshot
-rw-r--r--    1 esherman  staff  15703578 Jul 25 15:09 afterFailure8.heapsnapshot
-rw-r--r--    1 esherman  staff  15710169 Jul 25 15:09 afterFailure9.heapsnapshot
```

Without the flow.wait
```
-rw-r--r--    1 esherman  staff   15976565 Jul 25 15:11 a_beforeFailure.heapsnapshot
-rw-r--r--    1 esherman  staff   70591109 Jul 25 15:10 afterFailure0.heapsnapshot
-rw-r--r--    1 esherman  staff  130996152 Jul 25 15:10 afterFailure1.heapsnapshot
-rw-r--r--    1 esherman  staff   74629501 Jul 25 15:10 afterFailure2.heapsnapshot
-rw-r--r--    1 esherman  staff   89390982 Jul 25 15:10 afterFailure3.heapsnapshot
-rw-r--r--    1 esherman  staff   89351152 Jul 25 15:10 afterFailure4.heapsnapshot
-rw-r--r--    1 esherman  staff   89428719 Jul 25 15:10 afterFailure5.heapsnapshot
-rw-r--r--    1 esherman  staff   89288946 Jul 25 15:10 afterFailure6.heapsnapshot
-rw-r--r--    1 esherman  staff   90315292 Jul 25 15:11 afterFailure7.heapsnapshot
-rw-r--r--    1 esherman  staff   89353936 Jul 25 15:11 afterFailure8.heapsnapshot
-rw-r--r--    1 esherman  staff   80478208 Jul 25 15:11 afterFailure9.heapsnapshot
```

**Review by shermaneric — DISMISSED** [review]

File: test/spec/protractor_sync_test.ts [hunk 1]
```diff
@@ -555,11 +555,13 @@ describe('Protractor extensions', () => {
...
 
     it('times out', createTest(() => {
-      let counter = 0;
+      let expectationMethodCheckCount = 0;
 
       expect(() =>
-        protractorSync.polledExpect(() => counter++, 100).toBeLessThan(0)
+        protractorSync.polledExpect(() => expectationMethodCheckCount++, 100).toBeLessThan(0)
       ).toThrowError(/Expected \d+ to be less than 0\./);
+      expect(expectationMethodCheckCount).toBeGreaterThan(9);
```
Comment (shermaneric) [inline test/spec/protractor_sync_test.ts]: Minor:  When I ran the this test 1000 times,  I received (albeit only twice) the error:
``` Expected 9 to be greater than 9.``` 

 Changing this to 8 I never saw the error in 1000 times.

**Comment by sharp-bits** [comment]
I should have listened to Nyquist. Thanks for the thorough validation.

**Review by mindywhitsitt — APPROVED** [review]

**Pull request merged**